### PR TITLE
refactor(SettingsContactEdit): code

### LINKED ISSUE #390: settings-contact-edit

### PR DESCRIPTION
**Pull request by samandarkuchkarov**
Fixes #390 

## Proposed Changes

-Change the colors from the constants to similar ones in Color
-Change the imports to absolute:
-Transferring the screen logic to components

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/settings-address-book-edit.tsx [hunk 1]
```diff
@@ -50,30 +50,30 @@ export const SettingsAddressBookEdit = memo(
...
+          i18nLabel={I18N.address}
           editable={false}
           value={initAddress}
         />
         {buttonType === 'save' ? (
-          <View style={page.buttonContainer}>
+          <View style={styles.buttonContainer}>
             <Button
               disabled={initName === inputName}
               title={getText(I18N.continue)}
```
Comment (gHashTag) [inline src/components/settings-address-book-edit.tsx:79]: `i18nTitle`

Comment (samandarkuchkarov) [reply src/components/settings-address-book-edit.tsx]: done

**Review by gHashTag — APPROVED** [review]

File: src/components/settings-contact-edit.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    const cancel = getText(I18N.cancel);
+    Alert.alert(delateContact, sure, [
+      {text: cancel, style: 'cancel'},
+      {
+        text: deleteWord,
+        style: 'destructive',
+        onPress: () => {
+          contacts.removeContact(address);
+          goBack();
+        },
```
Comment (vivalaakam) [inline src/components/settings-contact-edit.tsx:47]: move these callback to screen

File: src/components/settings-contact-edit.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  const isChanged = inputName !== name;
+
+  const onSubmit = (newName?: string) => {
+    if (isCreate) {
+      contacts.createContact(address, newName || inputName);
+    } else {
+      contacts.updateContact(address, newName || inputName);
+    }
+    goBack();
+  };
```
Comment (vivalaakam) [inline src/components/settings-contact-edit.tsx:33]: move on screen

File: src/components/settings-contact-edit.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import {useContacts, useTypedNavigation} from '@app/hooks';
+import {useTypedRoute} from '@app/hooks/use-typed-route';
+import {I18N, getText} from '@app/i18n';
+
+export const SettingsContactEdit = () => {
+  const {name, address, isCreate} =
+    useTypedRoute<'settingsContactEdit'>().params;
+
+  const contacts = useContacts();
+  const {goBack} = useTypedNavigation();
```
Comment (vivalaakam) [inline src/components/settings-contact-edit.tsx:18]: move on scree

File: src/components/settings-contact-edit.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      setActionSheetVisible(true);
+    } else {
+      goBack();
+    }
+  };
+
+  const onPressDiscard = () => {
+    setActionSheetVisible(false);
+    goBack();
+  };
```
Comment (vivalaakam) [inline src/components/settings-contact-edit.tsx:72]: move on screen

File: src/components/settings-contact-edit.tsx (hunk 4 above)
Comment (samandarkuchkarov) [reply src/components/settings-contact-edit.tsx]: done

File: src/components/settings-contact-edit.tsx (hunk 5 above)
Comment (samandarkuchkarov) [reply src/components/settings-contact-edit.tsx]: done

File: src/components/settings-contact-edit.tsx (hunk 3 above)
Comment (samandarkuchkarov) [reply src/components/settings-contact-edit.tsx]: done

File: src/components/settings-contact-edit.tsx (hunk 2 above)
Comment (samandarkuchkarov) [reply src/components/settings-contact-edit.tsx]: done

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**